### PR TITLE
🌎 Allow https proxy from env var

### PR DESCRIPTION
**Pull request by fwkoch**
This PR moves `fetch` to a function on `session` to centralize the logic, then adds `HTTPS_PROXY` consumption: #919

### CODE REVIEW

File: packages/myst-cli-utils/src/index.ts [hunk 1]
```diff
@@ -10,7 +10,6 @@ export {
 export { exec, makeExecutable } from './exec.js';
 export { clirun, tic } from './utils.js';
 export { isUrl } from './isUrl.js';
-export { Session, getSession } from './session.js';
```
Comment (fwkoch) [inline packages/myst-cli-utils/src/index.ts:13]: This super minimal session implementation isn't used anywhere (besides a single test file - easily replaced). I just removed it so we did not need to implement `fetch` in this package.

File: packages/myst-cli/src/build/html/index.ts [hunk 2]
```diff
@@ -111,7 +110,7 @@ export async function buildHtml(session: ISession, opts: any) {
   // Fetch all HTML pages and assets by the template
   await Promise.all(
     routes.map(async (page) => {
-      const resp = await fetch(page.url);
+      const resp = await session.fetch(page.url);
```
[the remark this answers is not in the document]
Comment (stevejpurves) [reply packages/myst-cli/src/build/html/index.ts:113]: yes sounds like we do? i.e. skip proxy requests for any localhost / 127.0.0.1 requests based on the url, and proxy all others?

**Comment by fwkoch** [comment]
I tested basic functionality running against a simple localhost proxy - all seems to work ok. Good enough to release and see how it goes with "real" proxies, I think.

**Review by rowanc1 — APPROVED** [review]
Looks good @fwkoch, merge away when you are ready!

**Comment by stevejpurves** [comment]
@fwkoch what did you use for your local proxy in the end (realizing here that a basic http server won't cut it, as it won't handle CONNECT)? I've had trouble getting it to connect to `http-configurable-proxy`

**Pull request merged**